specs/merge/fork-choice.md

```diff
@@ -0,0 +1,113 @@
+# Ethereum 2.0 The Merge
+
+**Notice**: This document is a work-in-progress for researchers and implementers.
+
+## Table of contents
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+  - [Helpers](#helpers)
+    - [`PowBlock`](#powblock)
+    - [`get_pow_block`](#get_pow_block)
+    - [`is_valid_transition_block`](#is_valid_transition_block)
+  - [Updated fork-choice handlers](#updated-fork-choice-handlers)
+    - [`on_block`](#on_block)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+This is the modification of the fork choice according to the executable beacon chain proposal.
+
+*Note*: It introduces the process of transition from the last PoW block to the first PoS block.
+
+### Helpers
+
+#### `PowBlock`
+
+```python
+class PowBlock(Container):
+    is_processed: boolean
+    is_valid: boolean
+    total_difficulty: uint256
```

Comment:
Yes, total difficulty falls into `> 2**74` interval today. The other potential way of handling this is to return an offset wrt some absolute total difficulty value. Current block's difficulty is around `2**64`, so we indeed may divide total difficulty by `2**20` without loss of generality.

Reply:
Also, it worth noting that `uint256` type is used by `Transaction` data structure to define several fields.

Reply:
ah, right. I don't thnk we'll be able to get around `value` with 256 granularity

I suppose beacon clients don't actually have to support arithmetic on these TX values because all arithmetic and validations happen in application layer so as long as they can serialize and deserialize, that's enough support